scripts/build_deb.py

```diff
@@ -0,0 +1,333 @@
+
+def get_deb_version(build_info: BuildInfo):
+
+    pattern = re.compile('^libsilkit \(([0-9]+)\.([0-9]+)\.([0-9]+)-([0-9]+).*')
+
+    changelog_path = build_info.silkit_pkg_path.expanduser() / 'debian/changelog'
+    logger.debug(f"Checking {changelog_path}")
+    if not changelog_path.expanduser().exists():
+        logger.error("Could not find Package Changelog! Exiting!")
+        cleanup(build_info, 64)
+
+    with open(changelog_path) as f:
+        for line in f:
+            result = re.match(pattern, line)
+            if result:
+                build_info.version = SilKitVersion(
+                    major=result.group(1),
+                    minor=result.group(2),
+                    patch=result.group(3),
+                    suffix=result.group(4))
+
+            logger.debug(f'SilKitVersion: {build_info.version}')
+            return
+
+def create_work_directory(build_info: BuildInfo) -> Path:
+
+    work_dir = Path.cwd() / build_info.work_dir;
+    logger.debug(f"Creating {work_dir} work dir!")
+
+    try:
+        Path.mkdir(work_dir, exist_ok=True)
+    except Exception as ex:
+        logger.error(f"While creating the workdir an error occured: {str(ex)}")
+        # No need for cleanup, nothing of value created yet
+        exit(64)
+
+def create_orig_tarball(build_info: BuildInfo):
+
+    silkit_version = build_info.version
+
+    if silkit_version == None:
+        logger.error("No valid SilKit Version found! Exiting!")
+        cleanup(build_info, 64)
+
+    tarball_name = 'libsilkit_{}.{}.{}.orig.tar.gz'.format(
+            silkit_version.major,
+            silkit_version.minor,
+            silkit_version.patch)
+    try:
+        subprocess.run(['tar', '--exclude=.git', '-czf', build_info.work_dir/tarball_name,
+                        '-C', build_info.silkit_info.silkit_source_path, '.'], check=True)
+    except Exception as ex:
+        logger.error(f"While creating the orig tarball, an error occured!\n{str(ex)}")
+        cleanup(build_info, 64)
+
+def copy_debian_dir(build_info: BuildInfo):
+
+    # Copy debian dir
+    try:
+        shutil.copytree(build_info.silkit_pkg_path.expanduser() / 'debian/', build_info.work_dir / 'sil-kit/debian')
+    except Exception as ex:
+        logger.error("Could not copy the debian dir into the sil kit source dir! Exiting")
+        logger.debug(f"Python Exception: {str(ex)}")
+        cleanup(build_info, 64)
+
+def parse_platform(platform: str) -> str:
+
+    dist, version = platform.split('-')
+    logger.info(f"Building for {dist} version {version}")
+
+    return version
+
+
+def get_build_flags(ubuntu_version: str) -> BuildFlags:
+
+    logger.info(f"Building for platform: {ubuntu_version}")
+    if ubuntu_version == "20.04":
+        return BuildFlags(
+                add_platform_flags="",
+                add_debuild_flags="-d --prepend-path=/opt/vector/bin",
+                c_compiler="clang-10",
+                cxx_compiler="clang++-10")
```

Comment:
is there a reason why we use clang? isn't GCC the native default compiler on ubuntu?

Reply:
I talked with Daniel about this and the idea was to maybe standardize on a clang version cause they  can be trivially installed on any Ubuntu release that is still supported.